packages/stryker-webpack-transpiler/src/compiler/ChunkSorter.ts

```diff
@@ -0,0 +1,57 @@
+import * as _ from 'lodash';
+const toposort = require('toposort');
+
+export interface Chunk {
+  id: string;
+  files: string[];
+  parents?: Array<string | Chunk>;
+}
+
+/**
+ * A plugin that can sort the webpack files given there dependencies
+ * so that they can be loaded in the browser in the correct order.
+ * 
+ * Given this dependency graph
+ *       A   D
+ *      / \ /
+ *     B   C
+ * 
+ * The order might be:
+ * D, A, B, C
+ * 
+ * @see https://www.npmjs.com/package/toposort#usage
+ */
+export default class ChunkSorter {
+
+  sortedFileNames(allChunks: Chunk[]) {
+    const sortedChunks = this.sort(allChunks);
+    return _.flatMap(sortedChunks, chunk => chunk.files);
+  }
+
+  private sort(allChunks: Chunk[]): Chunk[] {
+    // We build a map (chunk-id -> chunk) for faster access during graph building.
+    const chunkMap: { [chunkId: string]: Chunk } = {};
+    allChunks.forEach((chunk) => {
+      chunkMap[chunk.id] = chunk;
+    });
+    const edges: [Chunk, Chunk][] = [];
+
+    allChunks.forEach(chunk => {
+      if (chunk.parents) {
+        // Add an edge for each parent (parent -> child)
```

Comment:
This code may be clearer if you explain the *parent -> child* relationship when you create the edges array.

Reply:
Good call. I've changed it in this:

```ts
   // Build the *edges* of the graph. 
    // It will contain an entry per parent -> child relationship
    const edges: [Chunk, Chunk][] = [];
```